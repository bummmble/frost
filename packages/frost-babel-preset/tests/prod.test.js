const buildPreset = require('../dist/index');
const { fixtures, titles, check } = require('./helpers');

const options = buildPreset(null, {
  modules: false,
  sourceMaps: false,
  compression: true,
  env: 'production',
});

fixtures.forEach((filename, index) => {
  test(titles[index], () => check(filename, options));
});
