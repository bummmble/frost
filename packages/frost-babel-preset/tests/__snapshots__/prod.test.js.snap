// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`deadcode 1`] = `
"const TEST = !1;
TEST, !1;"
`;

exports[`es2015: classes 1`] = `
"export class Component {
  main() {}
}"
`;

exports[`es2015: generator 1`] = `
"function* tester() {
  for (let a = 0;;) yield a++;
}"
`;

exports[`es2015: set 1`] = `"new Set([1, 2, 3]);"`;

exports[`es2016: exponentiation 1`] = `"var test = 4;"`;

exports[`esnext: async 1`] = `
"import _default from \\"nodent-runtime\\";
function wait() {
  return new Promise(function (a, b) {
    return test().then(function () {
      return a();
    }.$asyncbind(this, b), b);
  }.$asyncbind(this));
}"
`;

exports[`esnext: class properties 1`] = `
"export class Comp {
  constructor() {
    this.onClick = () => {};
  }

}"
`;
